From: Keith Packard <keithp@keithp.com>
Date: Tue, 26 Sep 2023 22:17:39 -0700
Subject: ports/qemu-{arm,riscv}: Support updated picolibc semihosting args

Picolibc's argument processing has changed with version 1.8.5 so that
the full command line, including program name, must be provided in the
'arg=' parameter to qemu.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
 ports/qemu-arm/snek-arm.in     | 4 ++--
 ports/qemu-riscv/snek-riscv.in | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/ports/qemu-arm/snek-arm.in b/ports/qemu-arm/snek-arm.in
index 0b1f832..a427287 100644
--- a/ports/qemu-arm/snek-arm.in
+++ b/ports/qemu-arm/snek-arm.in
@@ -1,6 +1,6 @@
 #!/bin/sh
-here=`dirname $0`
-args=""
+here=`dirname "$0"`
+args=`basename "$0"`
 extra=""
 for i in "$@"; do
     case "$i" in
diff --git a/ports/qemu-riscv/snek-riscv.in b/ports/qemu-riscv/snek-riscv.in
index 6fee1af..9863593 100644
--- a/ports/qemu-riscv/snek-riscv.in
+++ b/ports/qemu-riscv/snek-riscv.in
@@ -1,6 +1,6 @@
 #!/bin/sh
-here=`dirname $0`
-args=""
+here=`dirname "$0"`
+args=`basename "$0"`
 extra=""
 for i in "$@"; do
     case "$i" in
